pkg/driver/provisioner_server_impl.go

```diff
@@ -130,11 +217,19 @@ func (s *ProvisionerServer) DriverCreateBucket(ctx context.Context,
 	if err != nil {
 		var bucketAlreadyExists *s3types.BucketAlreadyExists
 
+		// Handle BucketAlreadyExists as a special case (no retry)
 		if errors.As(err, &bucketAlreadyExists) {
 			klog.V(c.LvlInfo).InfoS("Bucket already exists", "bucketName", bucketName)
 			return nil, status.Errorf(codes.AlreadyExists, "Bucket already exists: %s", bucketName)
+		}
+
+		// Use mapping for other errors
+		grpcStatus := mapErrorToGRPCStatus(err, createBucketErrorMappings, bucketName)
+		if grpcStatus.Code() == codes.InvalidArgument {
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid bucket configuration: %s", bucketName)
+		} else if grpcStatus.Code() == codes.PermissionDenied {
+			return nil, status.Errorf(codes.PermissionDenied, "Access denied for bucket: %s", bucketName)
 		} else {
-			klog.ErrorS(err, "Failed to create bucket", "bucketName", bucketName)
 			return nil, status.Error(codes.Internal, "Failed to create bucket")
```

Comment:
why not put the bucketname in the string here ?

Reply:
I was not done making changes, and went to a meeting

Reply:
I made the change in https://github.com/scality/cosi-driver/pull/188/commits/dfcea6f15764f7dbce752a12c01f69fd2ff48625